pycbc/workflow/core.py

```diff
@@ -638,19 +639,32 @@ def execute_node(self, node, verbatim_exe = False):
     
     @staticmethod
     def set_job_properties(job, output_map, staging_site=None):
-        job.addArguments('-Dpegasus.dir.storage.mapper.replica.file=%s' % output_map) 
+        job.addArguments('-Dpegasus.dir.storage.mapper.replica.file=%s' % 
+                         os.path.basename(output_map))
+        job.uses(Pegasus.DAX3.File(os.path.basename(output_map)), 
+                 link=Pegasus.DAX3.Link.INPUT)
         job.addArguments('-Dpegasus.dir.storage.mapper.replica=File') 
-        job.addArguments('--cache %s' % os.path.join(os.getcwd(), '_reuse.cache')) 
+
         job.addArguments('--output-site local')     
         job.addArguments('--cleanup inplace')
         job.addArguments('--cluster label,horizontal')
         job.addArguments('-vvv')
+
+        # FIXME _reuse_cache needs to be fixed to use PFNs properly. This will
+        # work as pegasus-plan is currently invoked on the local site so has
+        # access to a file in os.getcwd() but this code is fragile.
+        job.addArguments('--cache %s' % os.path.join(os.getcwd(), '_reuse.cache'))
+
         if staging_site:
             job.addArguments('--staging-site %s' % staging_site)
             
     def save(self, filename=None, output_map=None, staging_site=None):
         if output_map is None:
             output_map = self.output_map
+            if self.in_workflow is not False:
```

Comment:
@duncan-brown I think I'm confused by these lines. How does the parent workflow know this is the map file to use when planning the sub-workflow? I see no connection between the two.